app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,38 @@
 class PostsController < ApplicationController
-  def index
-    @user = User.find(params[:user_id])
-    @posts = Post.where(author: @user).order(id: :asc)
+  @post = Post.find(params[:id])
+  @comments = @post.comments
+  @likes = @post.likes
+end
+
+def new
+  @post = Post.new
+  respond_to do |format|
+    format.html { render :new, locals: { post: @post } }
   end
+end
 
-  def show
-    @post = Post.find(params[:id])
-    @comments = @post.comments
+def create
+  @post = Post.new(strong_params)
+  @post.author = current_user
+  respond_to do |format|
+    format.html do
+      if @post.save
+        flash[:success] = 'Post saved successfully'
+        redirect_to users_url
+      else
+        flash.now[:error] = 'Error: Post could not be saved'
+        render :new, locals: { post: @post }
+      end
+    end
   end
 end
+
+  private
+
+# Using a private method to encapsulate the permissible parameters
+# is just a good pattern since you'll be able to reuse the same
+# permit list between create and update. Also, you can specialize
+# this method with per-user checking of permissible attributes.
+def strong_params
+  params.require(:post).permit(:title, :text, :author)
+end
```

Comment:
- [OPTIONAL] You could name here differently using the model name like this `post_params`
- [ ] The allowed parameters are just `title` and `text`, `author` is a field coming from the foreign key as the post is created on behalf of a user (author)